plugins/nf-azure/src/main/nextflow/cloud/azure/file/AzBashLib.groovy

```diff
@@ -31,30 +31,67 @@ class AzBashLib extends BashFunLib<AzBashLib> {
 
     private String blockSize = AzCopyOpts.DEFAULT_BLOCK_SIZE
     private String blobTier = AzCopyOpts.DEFAULT_BLOB_TIER
-
+    private String putMD5 = ''
+    private String checkMD5 = AzCopyOpts.DEFAULT_CHECK_MD5
+    private String overwrite = AzCopyOpts.DEFAULT_OVERWRITE
+    //private String outputLevel = AzCopyOpts.DEFAULT_OUTPUT_LEVEL
+    //private String requestTryTimeout = AzCopyOpts.DEFAULT_AZCOPY_REQUEST_TRY_TIMEOUT
 
     AzBashLib withBlockSize(String value) {
-        if( value )
+        if (value)
             this.blockSize = value
         return this
     }
 
     AzBashLib withBlobTier(String value) {
-        if( value )
+        if (value)
             this.blobTier = value
         return this
     }
 
+    AzBashLib withPutMD5(Boolean value) {
+        this.putMD5 = value ? '--put-md5' : ''
+        return this
+    }
+
+    AzBashLib withCheckMD5(String value) {
+        if (value)
+            this.checkMD5 = value
+        return this
+    }
+
+    AzBashLib withOverwrite(String value) {
+        if (value)
+            this.overwrite = value
+        return this
+    }
+
+    /* //DISABLED TILL WE UPGRADE TO azcopy-v10.16.0
+    AzBashLib withOutputLevel(String value) {
+        if (value)
+            this.outputLevel = value
+        return this
+    }
+
+    AzBashLib withRequestTryTimeout(String value) {
+        if (value)
+            this.requestTryTimeout = value
+        return this
+    }
+    */
 
+    // Custom env variables are prefixed with underscore(_)
     protected String setupAzCopyOpts() {
         """
-        # custom env variables used for azcopy opts
-        export AZCOPY_BLOCK_SIZE_MB=${blockSize}
-        export AZCOPY_BLOCK_BLOB_TIER=${blobTier}
+        # Env variables used for azcopy opts
+        export _AZCOPY_BLOCK_SIZE_MB=${blockSize}
+        export _AZCOPY_BLOCK_BLOB_TIER=${blobTier}
+        export _AZCOPY_PUT_MD5=${putMD5}
+        export _AZCOPY_CHECK_MD5=${checkMD5}
+        export _AZCOPY_OVERWRITE=${overwrite}
```

Comment:
Why they are prefixed with `_` ?

Reply:
@pditommaso , I did this to distinguish the custom env vars with the actual env vars such as `AZCOPY_REQUEST_TRY_TIMEOUT` which we had to remove now and would be a part of next iteration.